packages/material-ui/src/ButtonBase/ButtonBase.test.js

```diff
@@ -117,6 +117,31 @@ describe('<ButtonBase />', () => {
       expect(button).to.have.property('nodeName', 'SPAN');
       expect(button).to.have.attribute('href', 'https://google.com');
     });
+
+    it('should allow to set type prop when component is not a button', () => {
+      const MockButton = React.forwardRef((props, ref) => (
+        <button data-testid="mockButton" {...props} ref={ref} />
+      ));
+
+      const { getByRole } = render(
+        // @ts-ignore
+        <ButtonBase component={MockButton} type="submit">
+          Hello
+        </ButtonBase>,
+      );
+      const button = getByRole('button');
+      expect(button).to.have.attribute('type', 'submit');
+    });
+
+    it('should not set type prop when component is an anchor', () => {
+      const { getByText } = render(
+        <ButtonBase component="a" type="submit">
```

Comment:
I think that we could expect it to be set, we could consider that developers are forcing it. From what I recall, the most important property we were looking for is not to set the type by default.

Reply:
That makes sense, was afraid that this was made on purpose. Let me try to think about a different solution here. Thanks for taking a quick look, appreciate it.

Reply:
@oliviertassinari I've reused the approach that you folks have taken in v5. Hope it makes sense.